src/networkapi.jl

```diff
@@ -574,6 +574,63 @@ function deficiency(ns, ig, lc)
    LG.nv(ig) - length(lc) - rank(matrix(FlintZZ,ns))
 end
 
+	
+
+"""
+	`subnetworkmapping` is an intermediary function
+	required in `subnetworks(network, linkage_classes)`
+"""
+# intermediary function for subnetwork of a particular linkage class
+function subnetworkmapping(linkageclass::Vector{Int64}, r, reacmap, pars)
+   rxind = unique(vcat([map(first, reacmap[j]) for j in 1:length(linkageclass)]...))
+   specs = unique(vcat([[r[rxind][j].substrates ; r[rxind][j].products] for j in 1:length(r[rxind])]...))
```

Comment:
Similar comment to above. I'd try to simplify this down. Maybe use a `Set` again and then convert to a vector afterwards? (You wouldn't need to sort this one.)

Reply:
> Similar comment to above. I'd try to simplify this down. Maybe use a `Set` again and then convert to a vector afterwards? (You wouldn't need to sort this one.)

like this ?
```julia
rxind = collect(Set(vcat([map(first, reacmap[j]) for j in 1:length(linkageclass)]...)))
   specs = collect(Set(vcat([[r[rxind][j].substrates ; r[rxind][j].products] for j in 1:length(r[rxind])]...)))

```

Reply:
The point is to avoid all the allocations of intermediary vectors. Just directly insert into the Set, you can always use loops if needed…

Reply:
> The point is to avoid all the allocations of intermediary vectors. Just directly insert into the Set, you can always use loops if needed…

i wll hv to think more about this. for now have made updates that were easy to work with